### DISCUSSION TITLE
Adding support for user defined parameterized mappings

### DISCUSSION THREAD

**Discussion by ldeluigi** [riok/mapperly#444] [Ideas]
Take the following example:
```csharp
public record MembershipDto(
    Guid Id,
    Option<UserDto> LinkedUser);

[Mapper]
public static partial class MembershipDtoMapper
{
    public static Option<Y> MapOptions<X, Y>(Option<X> src, Func<X, Y> mapping) => src.Map(mapping);

    public static partial UserDto MapFrom(UserSnapshot src);

    public static partial MembershipDto MapFrom(MembershipSnapshot src);
}
```

I'd like mapperly to detect and pick up `MapOptions` and call it by passing the source and a mapping from X to Y.
`Option<X>` is a custom class representing an optional value, and we can't use Mapperly because there is no way to tell Mapperly how to map `Option<X>` to `Option<Y>`

**Comment by latonz** [comment]
Thank you for opening this discussion and bringing up this idea, I created https://github.com/riok/mapperly/issues/451 to track this. Implementing this needs a lot of effort, and therefore may take a long time.
